zephyr/wrapper.c

```diff
@@ -303,6 +315,35 @@ uint64_t platform_timer_get(struct timer *timer)
 #elif CONFIG_SOC_SERIES_INTEL_ADSP_BROADWELL || CONFIG_LIBRARY
 	// FIXME!
 	return 0;
+#elif CONFIG_IMX
+	/* For i.MX use Xtensa timer, as we do now with SOF */
+	uint64_t time = 0;
+	uint32_t low;
+	uint32_t high;
+	uint32_t ccompare;
+
+	if (timer->id >= ARCH_TIMER_COUNT)
+		goto out;
+
+	ccompare = xthal_get_ccompare(timer->id);
+
+	/* read low 32 bits */
+	low = xthal_get_ccount();
+
+	/* check and see whether 32bit IRQ is pending for timer */
+	if (arch_interrupt_get_status() & (1 << timer->irq) && ccompare == 1) {
+		/* yes, overflow has occurred but handler has not run */
+		high = timer->hitime + 1;
+	} else {
+		/* no overflow */
+		high = timer->hitime;
+	}
+
+	time = ((uint64_t)high << 32) | low;
+
+out:
+
+	return time;
 #else
```

Comment:
My understanding here is that we want an implementation for platform_timer_get() that doesn't use XTOS API at all but it uses APIs from Zephyr. 

@lgirdwood is this true?

Reply:
Seeing the "TODO: align with 64bit Zephyr timers when they are upstream." from [here](https://github.com/thesofproject/sof/blob/main/zephyr/wrapper.c#L276), for now, I've kept the Xtensa timer.

From my understanding Intel does the same with CAVS platforms supported in Zephyr - see [here](https://github.com/thesofproject/sof/blob/main/zephyr/wrapper.c#L308)

But, I'll let @lgirdwood comment also.

Reply:
yes, we are moving to use Zephyr native APIs through out the codebase. @kv2019i @marc-hb and @lyakh are working on this. The intention is to gradually remove code from zephyr/wrapper.c and have an xtos/wrapper.c (and eventually deprecate xtos).